application/templates/secretProviderClass.yaml

```diff
@@ -0,0 +1,23 @@
+{{- if .Values.secretProviderClass.enabled }}
+apiVersion: secrets-store.csi.x-k8s.io/v1alpha1
+kind: SecretProviderClass
+metadata:
+  name: {{.Values.secretProviderClass.name}}
+spec:
+  provider: {{.Values.secretProviderClass.provider}}
+  parameters:
+    vaultAddress: {{.Values.secretProviderClass.parameters.vaultAddress}}
+    roleName: {{.Values.secretProviderClass.parameters.roleName}}
+    objects: 
+    {{- if .Values.secretProviderClass.parameters.objects }}
+    {{ toYaml .Values.secretProviderClass.parameters.objects | indent 4 }}
+    {{- end }}
+  secretObjects:
+  {{- if .Values.secretProviderClass.secretObjects }}
```

Comment:
this should come line above than this

Reply:
??

Reply:
if condition should come before line "secretObjects". so if secretObjects are not defined in values, this field dont come at all

Reply:
Done